src/main/java/com/yahoo/bullet/bql/parser/ParsingException.java

```diff
@@ -43,15 +42,15 @@ public ParsingException(String message) {
         this(message, null, 1, 0);
     }
 
-    /**
-     * Constructor to initialize ParsingException with a message and {@link NodeLocation}.
-     *
-     * @param message      The error message to be associated with the ParsingException.
-     * @param nodeLocation The {@link NodeLocation} that contains line number and char position in line.
-     */
-    public ParsingException(String message, NodeLocation nodeLocation) {
```

Comment:
We should change this to an Error that's returned when parsing rather than an exception and should contain a contextual location in addition to just line number and character position if possible

Reply:
Ok, we'll probably have to put back node location then. cough